flink-table/flink-table-common/src/main/java/org/apache/flink/table/utils/PrintUtils.java

```diff
@@ -176,14 +229,101 @@ public static void printAsTableauForm(
 		return colWidths;
 	}
 
-	public static String genBorderLine(int[] colWidths) {
-		StringBuilder sb = new StringBuilder();
-		sb.append("+");
-		for (int width : colWidths) {
-			sb.append(EncodingUtils.repeat('-', width + 1));
-			sb.append("-+");
+	/**
+	 * Try to infer column width based on column types. In streaming case, we will have an
```

Comment:
we try to avoid the terminology `streaming` and `batch` in the code base, instead we call it `unbounded` or `bounded`